compare_ce_spg_sample_file_with_original.py

```diff
@@ -0,0 +1,62 @@
+import argparse
+import csv
+
+
+def compare_files(old_file_path, new_file_path):
+    old_file_by_uprn = {}
+
+    problems_found = False
+
+    with open(old_file_path) as old_file:
+
+        old_file_reader = csv.DictReader(old_file, delimiter=',')
+
+        for count, sample_row in enumerate(old_file_reader, 1):
+            old_file_by_uprn[f'{sample_row["UPRN"]}'] = sample_row
+
+    with open(new_file_path) as new_file:
+
+        new_file_reader = csv.DictReader(new_file, delimiter=',')
+
+        unique_uprns = set()
+
+        for count, sample_row in enumerate(new_file_reader, 1):
+            matching_sample_row = old_file_by_uprn.get(sample_row['UPRN'])
+            if not matching_sample_row:
+                print(f'Could not find UPRN {sample_row["UPRN"]} on row {count + 1}')
+                problems_found = True
+            else:
+
+                if sample_row["UPRN"] in unique_uprns:
+                    print(f'Duplicate line on row {count + 1}')
+                    problems_found = True
+
+                unique_uprns.add(sample_row['UPRN'])
+
+                for row_key in list(sample_row.keys()):
+                    if sample_row[row_key] != matching_sample_row[row_key] \
+                            and row_key != 'FIELDCOORDINATOR_ID' and row_key != 'FIELDOFFICER_ID':
+                        print(f'Found invalid data in column {row_key}, row {count + 1}: {sample_row[row_key]}')
+                        problems_found = True
+
+    return problems_found
```

Comment:
if you returned a list of the problems found, it'd be easier to unit test

Reply:
I don't think it's returning a list of problems. It seems to be returning a mish-mash of different data fragments. Almost impossible to unit test properly